c/driver/postgresql/connection.cc

```diff
@@ -185,6 +186,10 @@ class PqGetObjectsHelper {
     na_error_ = {0};
   }
 
+  static const inline std::unordered_map<std::string, std::string> PgTableTypes = {
```

Comment:
I've attached this to PqGetObjectsHelper for now, though I think could also just be a static constant in the module

Reply:
nit: I don't think `inline` does much here, and I think coding convention would make this `kPgTableTypes`?

Reply:
Without the `inline` keyword I get the following error:

```
/home/willayd/clones/arrow-adbc/c/driver/postgresql/connection.cc:189:61: error: in-class initialization of static data member ‘const std::unordered_map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char> > {anonymous}::PqGetObjectsHelper::PgTableTypes’ of non-literal type
  189 |   static const std::unordered_map<std::string, std::string> PgTableTypes = {
      |                                                             ^~~~~~~~~~~~
/home/willayd/clones/arrow-adbc/c/driver/postgresql/connection.cc:191:79: error: temporary of non-literal type ‘const std::unordered_map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char> >’ in a constant expression
  191 |       {"toast_table", "t"}, {"foreign_table", "f"}, {"partitioned_table", "p"}};
```

AFAICT this is a C++17 feature

https://stackoverflow.com/a/38043566/621736

Reply:
I think you can just put it outside the class then (shouldn't matter since this is in a cc file and not a header). Plus we've been trying to avoid C++17 for R's sake (though apparently we don't enforce that!), since some R distributions still ship with only C++11 support

Reply:
Ah that's great info. Arrow core is C++17 right? Thought that was the case maybe misinterpreted

Reply:
Arrow C++ is C++17 indeed; I think what happened here is that I intended for C++17 (or really, didn't really think about it), but we didn't end up using anything from C++14 or 17 at all (except in test code), and it turned out to be more convenient to stick to C++11 in non-test code for now to help R. @paleolimbot can explain the R part better than I.

Reply:
I filed https://github.com/apache/arrow-adbc/issues/815 to make sure we set this explicitly